src/optional.coffee

```diff
@@ -28,7 +30,7 @@ None = {
 }
 
 toOption = (v) ->
-  if v instanceof Some or v == None
+  if v?._isSome or v == None
```

Comment:
Shouldn't `None` case also use something else than object identity for comparison?